labs/lab-2-javascript-game/javascript.js

```diff
@@ -0,0 +1,125 @@
+
+window.addEventListener("load", () => {
+  document.getElementById("initialize").addEventListener("click", startGame);
+  document.getElementById("attack").addEventListener("click", attack);
+  document.getElementById("heal").addEventListener("click", heal);
+  document.getElementById("retreat").addEventListener("click", retreat);
+});
+
+class Player {
+  constructor(name, health) {
+    this.name = name;
+    this.score = health;
+    this.heals = 2;
+    this.wins = 0;
+  }
+  attackDamage() {
+    this.score = this.score - getDamage();
+  }
+  healing() {
+    this.score = this.score + getHealing();
+
+  }
+}
+var enemy;
+var user;
+
+function startGame() {
+  var name = document.getElementById("first-input").value;
+  var enemyName = document.getElementById("second-input").value;
+  user = new Player(name, 40);
+  enemy = new Player(enemyName, 10);
+  console.log(user);
+  console.log(enemy);
+  enemyResults();
+  userResults();
+  hideBattle();
+}
+
+
+function userResults(){
+  document.getElementById("player-health").style.width = (playerHealthRemain()).toString() + "%";
+  document.getElementById("heal-count").style.width = (userHealsRemain()).toString() + "%";
+  document.getElementById("first-input").value;
+  document.getElementById("player-name").innerText = user.name + " has " + user.score + " health";
+}
+
+function enemyResults(){
+  document.getElementById("enemy-health").style.width = (enemyHealthRemain()).toString() + "%";
+  document.getElementById("second-input").value;
+  document.getElementById("enemy-name").innerText = enemy.name + " has " + enemy.score + " health";
+}
+
+function attack() {
+  user.attackDamage();
+  enemy.attackDamage();
+  battleTally();
+  enemyResults();
+  userResults();
+}
+
+function heal() {
+  if (user.heals > 0) {
+    user.healing();
+    user.heals = user.heals - 1;
+    document.getElementById("heal-count").style.width = (userHealsRemain()).toString() + "%";
+  }
```

Comment:
You don't update the `userResults`, here, so healing doesn't look like it worked until after the next attack.